omod/src/main/java/org/openmrs/web/controller/PortletController.java

```diff
@@ -183,7 +188,13 @@ public ModelAndView handleRequest(HttpServletRequest request, HttpServletRespons
 					
 					// add encounters if this user can view them
 					if (Context.hasPrivilege(PrivilegeConstants.GET_ENCOUNTERS)) {
-						model.put("patientEncounters", Context.getEncounterService().getEncountersByPatient(p));
+						Integer limit = getLimitParameter(request, as, "dashboard.encounters.maximumNumberToShow");
+						Integer startIndex = getStartIndexParameter(request);
+						
+						model.put(
+						    "patientEncounters",
+						    Context.getEncounterService().getEncounters(p.getPatientIdentifier().getIdentifier(),
```

Comment:
We should be using `Context.getEncounterService()getEncounters(EncounterSearchCriteria encounterSearchCriteria)`